File: Samples/app/src/main/java/samples/linhtruong/com/app/AppModule.java
package samples.linhtruong.com.app;

import android.app.Application;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

/**
 * CLASS DESCRIPTION
 *
 * @author linhtruong
 * @date 1/4/17 - 14:40.
 * @organization VED
 */

@Module
public class AppModule {

    private final Application mApplication;

    public AppModule(Application application) {
        mApplication = application;
    }

    @Provides
    @Singleton
    Application providesContext() {
        return mApplication;
    }
}

File: BaseProject/app/src/main/java/com/linhtruong/sample/core/di/module/AppModule.kt
package news.agoda.com.sample.di.module

import android.app.Application
import dagger.Module
import dagger.Provides
import com.linhtruong.sample.core.platform.App
import javax.inject.Singleton


/**
 * @author linhtruong
 */
@Module
class AppModule(private val application: App) {
    @Provides
    @Singleton
    fun provideApplicationContext(): Application = application
}